src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationMiddleware.cs

```diff
@@ -42,6 +43,7 @@ public class OpenIdConnectAuthenticationMiddleware : AuthenticationMiddleware<Op
             [NotNull] IDataProtectionProvider dataProtectionProvider,
             [NotNull] ILoggerFactory loggerFactory,
             [NotNull] IUrlEncoder encoder,
+            [NotNull] IDistributedCache cache,
```

Comment:
I'm not positive but I think you can just do make it optional by adding cache = null.  But I'm not sure this works on all DI implementations consistently...